src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Migrate/ProjectJsonToPackageRefMigrator.cs

```diff
@@ -73,23 +79,7 @@ public static class ProjectJsonToPackageRefMigrator
             dependencies.AddRange(packageSpec.Dependencies);
             foreach (var dependency in dependencies)
             {
-                var includeFlags = dependency.IncludeType;
-                var privateAssetsFlag = dependency.SuppressParent;
-                var metadataElements = new List<string>();
-                var metadataValues = new List<string>();
-                if (includeFlags != LibraryIncludeFlags.All)
-                {
-                    metadataElements.Add("IncludeAssets");
-                    metadataValues.Add(LibraryIncludeFlagUtils.GetFlagString(includeFlags).Replace(',', ';'));
-                }
-
-                if (privateAssetsFlag != LibraryIncludeFlagUtils.DefaultSuppressParent)
-                {
-                    metadataElements.Add("PrivateAssets");
-                    metadataValues.Add(LibraryIncludeFlagUtils.GetFlagString(privateAssetsFlag).Replace(',', ';'));
-                }
-
-                await project.InstallPackageWithMetadataAsync(dependency.Name, dependency.LibraryRange.VersionRange, metadataElements, metadataValues);
+                await project.ProjectServices.References.AddOrUpdatePackageReferenceAsync(dependency, CancellationToken.None);
```

Comment:
Good point! I planned to limit scope of the change here. I'll consider propagating `CancellationToken` up to the entry point.